### ISSUE TITLE
Eshell key binding: C-d (eshell-life-is-too-much)

### ISSUE THREAD

**Issue by microamp** [syl20bnr/spacemacs#4630]
Currently, `C-d` is bound to `eshell-life-is-too-much`. In `packages.el` (under `layers/shell`), you can see

```emacs-lisp
(add-hook 'eshell-mode-hook
         #'(lambda ()
             (define-key eshell-mode-map (kbd "C-l") 'eshell/clear)
             (define-key eshell-mode-map (kbd "C-d") 'eshell-life-is-too-much)))
```

`eshell-life-is-too-much` kills/buries the Eshell buffer even when there is a character left to delete. Ideally, `C-d` would either 1. delete a character forward, or 2. kills/buries the buffer depending on whether there's a character left to delete or not. This is, IIRC, consistent with what other shells do. I think `eshell-delchar-or-maybe-eof` (in `esh-rebind.el.gz`) does what I want.

Any thoughts?

**Comment by CestDiego** [comment]
Can you do a PR to represent what you want?

**Comment by robbyoconnor** [comment]
Do it against `develop` please. [peruse the contribution guidelines](https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org) so that we don't have to reject it :)

**Comment by microamp** [comment]
Thanks. I'll give it a go when I can unless someone else wants it done beforehand.

**Comment by microamp** [comment]
Fixed in #4681
